libs/analytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java

```diff
@@ -2579,124 +1221,13 @@ public static String getEventNameForStat(AnalyticsTracker.Stat stat) {
-                return "qrlogin_scanner_dismissed_camera_permission_denied";
-            case BLOGANUARY_NUDGE_MY_SITE_CARD_LEARN_MORE_TAPPED:
-                return "bloganuary_nudge_my_site_card_learn_more_tapped";
-            case BLOGANUARY_NUDGE_LEARN_MORE_MODAL_SHOWN:
-                return "bloganuary_nudge_learn_more_modal_shown";
-            case BLOGANUARY_NUDGE_LEARN_MORE_MODAL_DISMISSED:
-                return "bloganuary_nudge_learn_more_modal_dismissed";
-            case BLOGANUARY_NUDGE_LEARN_MORE_MODAL_ACTION_TAPPED:
-                return "bloganuary_nudge_learn_more_modal_action_tapped";
-            case SOTW_2023_NUDGE_POST_EVENT_CARD_SHOWN:
-                return "sotw_2023_nudge_post_event_card_shown";
-            case SOTW_2023_NUDGE_POST_EVENT_CARD_HIDE_TAPPED:
-                return "sotw_2023_nudge_post_event_card_hide_tapped";
-            case SOTW_2023_NUDGE_POST_EVENT_CARD_CTA_TAPPED:
-                return "sotw_2023_nudge_post_event_card_cta_tapped";
-            case DYNAMIC_DASHBOARD_CARD_SHOWN:
-                return "dynamic_dashboard_card_shown";
-            case DYNAMIC_DASHBOARD_CARD_TAPPED:
-                return "dynamic_dashboard_card_tapped";
-            case DYNAMIC_DASHBOARD_CARD_CTA_TAPPED:
-                return "dynamic_dashboard_card_cta_tapped";
-            case DYNAMIC_DASHBOARD_CARD_HIDE_TAPPED:
-                return "dynamic_dashboard_card_hide_tapped";
-            case DEEP_LINK_FAILED:
-                return "deep_link_failed";
-            case READER_DROPDOWN_MENU_OPENED:
-                return "reader_dropdown_menu_opened";
-            case READER_DROPDOWN_MENU_ITEM_TAPPED:
-                return "reader_dropdown_menu_item_tapped";
-            case SITE_MONITORING_SCREEN_SHOWN:
-                return "site_monitoring_screen_shown";
-            case OPENED_SITE_MONITORING:
-                return "opened_site_monitoring";
-            case SITE_MONITORING_TAB_SHOWN:
-                return "site_monitoring_tab_shown";
-            case SITE_MONITORING_TAB_LOADING_ERROR:
-                return "site_monitoring_tab_loading_error";
-            case NOTIFICATION_MENU_TAPPED:
-                return "notification_menu_tapped";
-            case NOTIFICATIONS_MARK_ALL_READ_TAPPED:
-                return "notifications_mark_all_read_tapped";
-            case NOTIFICATIONS_INLINE_ACTION_TAPPED:
-                return "notifications_inline_action_tapped";
-            case WEBVIEW_TOO_LARGE_PAYLOAD_ERROR:
-                return "webview_too_large_payload_error";
+            /*
+             * Please add a new case only if the new Stat's name in lower case does not match the expected event name.
+             * In that case you also need to add the event in the `AnalyticsTrackerNosaraTest.specialNames` map.
+             */
+            default:
+                return stat.name().toLowerCase(Locale.getDefault());
```

Comment:
NP: I'm not sure if this matters, but it might be worth considering hard-coding this to the US locale, as this is the language the enums are written in (in code). I.e.:

```suggestion
                return stat.name().toLowerCase(Locale.US);
```

Reply:
That's a good idea @mkevins 👍 
I changed the code to use `Locale.US` 🙇